src/pocketdb/web/PocketSystemRpc.cpp

```diff
@@ -92,6 +92,7 @@ namespace PocketWeb::PocketWebRpc
 
         UniValue proxies(UniValue::VARR);
         if (!WSConnections.empty()) {
```

Comment:
`WSConnections.empty()` is not atomic or thread safe or smth so here is still a small chance for data race here. Even if its implementation is atomic, there is no guarantee that map will not be changed in time gap before mutex lock. I mean even if map is not empty on this check, it could be empty after mutex lock. Moreover this check is absolutely unnecessary so i suggest remove it and described problem would be solved by no problem

Reply:
Good catch, I will submit a patch

Reply:
From looking at this more closely I think we should be okay with patch as-is.  empty() is okay for concurrent access according to https://www.cplusplus.com/reference/map/map/empty/

To verify I set a breakpoint on empty to step into the code and found it resolves to this:
```
_GLIBCXX_NODISCARD bool
      empty() const _GLIBCXX_NOEXCEPT
      { return _M_impl._M_node_count == 0; }
```